### PR TITLE
Fixed #27462 -- Clarifed difference in m2m_changed pk_set for add() vs remove().

### PR DESCRIPTION
**Pull request by carltongibson**
_Wording. Hmmm._ 

c.f. #10563.

### CODE REVIEW

File: tests/m2m_signals/tests.py [hunk 1]
```diff
@@ -69,6 +69,40 @@ def _initialize_signal_car(self, add_default_parts_before_set_signal=False):
...
+        """
+        pk_sets_sent = []
+
+        def handler(signal, sender, **kwargs):
+            if kwargs['action'] in ['pre_add', 'pre_remove']:
+                pk_sets_sent.append(kwargs['pk_set'])
+
+        models.signals.m2m_changed.connect(
+            handler, Car.default_parts.through
+        )
```
Comment (felixxm) [inline tests/m2m_signals/tests.py]: This can be single-lined.

File: tests/m2m_signals/tests.py [hunk 2]
```diff
@@ -69,6 +69,40 @@ def _initialize_signal_car(self, add_default_parts_before_set_signal=False):
...
+
+        self.vw.default_parts.add(self.wheelset)
+        self.vw.default_parts.add(self.wheelset)
+        self.vw.default_parts.add(self.wheelset)
+
+        self.vw.default_parts.remove(self.wheelset)
+        self.vw.default_parts.remove(self.wheelset)
+        self.vw.default_parts.remove(self.wheelset)
```
Comment (felixxm) [inline tests/m2m_signals/tests.py]: I think that two call of each method is enough.

File: tests/m2m_signals/tests.py [hunk 3]
```diff
@@ -69,6 +69,40 @@ def _initialize_signal_car(self, add_default_parts_before_set_signal=False):
...
+
+        expected_pk_sets = [{1}, set(), set(), {1}, {1}, {1}]
+        self.assertEqual(pk_sets_sent, expected_pk_sets)
+
+        models.signals.m2m_changed.disconnect(
+            handler, Car.default_parts.through
+        )
```
Comment (felixxm) [inline tests/m2m_signals/tests.py]: This can be single-lined.

File: docs/ref/signals.txt [hunk 4]
```diff
@@ -270,9 +270,16 @@ Arguments sent with this signal:
...
 
 ``pk_set``
-    For the ``pre_add``, ``post_add``, ``pre_remove`` and ``post_remove``
-    actions, this is a set of primary key values that have been added to
-    or removed from the relation.
+    For the ``pre_add`` and ``post_add`` actions, this this is a set of primary
+    key values that will be or have been added to the relation.
+
+    For the ``pre_remove`` and ``post_remove`` actions, this is a set of
+    primary key values that was was submitted to be removed from the relation,
```
[the remark this answers is not in the document]
Comment (carltongibson) [reply docs/ref/signals.txt]: Grrr... What had happened to my brain at this point I don't know. 😀

File: docs/ref/signals.txt [hunk 5]
```diff
@@ -270,9 +270,16 @@ Arguments sent with this signal:
...
+    regardless of whether they actually will be or have been so removed, i.e.
```
[the remark this answers is not in the document]
Comment (carltongibson) [reply docs/ref/signals.txt]: OK 👍

**Comment by carltongibson** [comment]
Cool. Thanks for the review @felixxm. I will go over it tomorrow. 👍

File: docs/ref/signals.txt [hunk 6]
```diff
@@ -270,9 +270,16 @@ Arguments sent with this signal:
...
+    non-existent values may be submitted but would not actually be removed.
+
+    (The difference here is because ``add()`` calls must filter existing values
+    in order to avoid an ``IntegrityError``.)
```
[the remark this answers is not in the document]
Comment (carltongibson) [reply docs/ref/signals.txt]: Yep. That's probably better. I will re-phase. Super.

**Pull request merged**